lib/http2/connection.c

```diff
@@ -541,7 +541,10 @@ static ssize_t expect_continuation_of_headers(h2o_http2_conn_t *conn, const uint
     }
 
     if (conn->_headers_unparsed->size + frame.length <= H2O_MAX_REQLEN) {
-        h2o_buffer_reserve(&conn->_headers_unparsed, frame.length);
+        if ((h2o_buffer_try_reserve(&conn->_headers_unparsed, frame.length)).base == NULL) {
```

Comment:
I do not think we need to use try_reserve here, because the maximum amount of memory being allocated is capped by `H2O_MAX_REQLEN` which is smaller than the mmap threshold.